pkg/openstack/loadbalancer_test.go

```diff
@@ -593,3 +595,296 @@ func Test_getIntFromServiceAnnotation(t *testing.T) {
+				Protocol:    pools.ProtocolHTTP,
+				LBMethod:    "SOURCE_IP_PORT",
+				Persistence: &pools.SessionPersistence{Type: "SOURCE_IP"},
+			},
+		},
+		{
+			name: "test for pool protocol UDP with proxy protocol disabled",
+			args: args{
+				protocol: "UDP",
+				svcConf: &serviceConfig{
+					keepClientIP:        true,
+					tlsContainerRef:     "tls-container-ref",
+					enableProxyProtocol: false,
+				},
+				lbaasV2: &LbaasV2{
+					LoadBalancer{
+						opts: LoadBalancerOpts{
+							LBProvider: "ovn",
+							LBMethod:   "SOURCE_IP_PORT",
+						},
+					},
+				},
+				service: &corev1.Service{
+					Spec: corev1.ServiceSpec{
+						SessionAffinity: corev1.ServiceAffinityClientIP,
+					},
+				},
+			},
+			want: pools.CreateOpts{
+				Protocol:    pools.ProtocolHTTP,
+				LBMethod:    "SOURCE_IP_PORT",
+				Persistence: &pools.SessionPersistence{Type: "SOURCE_IP"},
+			},
+		},
+		{
+			name: "test for session affinity none",
+			args: args{
+				protocol: "TCP",
+				svcConf: &serviceConfig{
+					keepClientIP:    true,
+					tlsContainerRef: "tls-container-ref",
+				},
+				lbaasV2: &LbaasV2{
+					LoadBalancer{
+						opts: LoadBalancerOpts{
+							LBProvider: "ovn",
+							LBMethod:   "SOURCE_IP_PORT",
+						},
+					},
+				},
+				service: &corev1.Service{
+					Spec: corev1.ServiceSpec{
+						SessionAffinity: corev1.ServiceAffinityNone,
+					},
+				},
+			},
+			want: pools.CreateOpts{
+				Protocol:    pools.ProtocolHTTP,
+				LBMethod:    "SOURCE_IP_PORT",
+				Persistence: nil,
+			},
+		},
+		{
+			name: "test for session affinity client ip",
+			args: args{
+				protocol: "TCP",
+				svcConf: &serviceConfig{
+					keepClientIP:    true,
+					tlsContainerRef: "tls-container-ref",
+				},
+				lbaasV2: &LbaasV2{
+					LoadBalancer{
+						opts: LoadBalancerOpts{
+							LBProvider: "ovn",
+							LBMethod:   "SOURCE_IP_PORT",
+						},
+					},
+				},
+				service: &corev1.Service{
+					Spec: corev1.ServiceSpec{
+						SessionAffinity: corev1.ServiceAffinityClientIP,
+					},
+				},
+			},
+			want: pools.CreateOpts{
+				Protocol:    pools.ProtocolHTTP,
+				LBMethod:    "SOURCE_IP_PORT",
+				Persistence: &pools.SessionPersistence{Type: "SOURCE_IP"},
+			},
+		},
+	}
+
+	for _, tt := range tests {
+		t.Run(tt.name, func(t *testing.T) {
+			got := tt.args.lbaasV2.buildPoolCreateOpt(tt.args.protocol, tt.args.service, tt.args.svcConf)
+			assert.Equal(t, got, tt.want)
+		})
+	}
+}
+
+func Test_getSubnetID(t *testing.T) {
+	type args struct {
+		svcConf *serviceConfig
+		service *corev1.Service
+		lbaasV2 *LbaasV2
+	}
+	tests := []struct {
+		name        string
+		args        args
+		want        string
+		expectedErr error
+	}{
+		{
+			name: "test get subnet from service annotation",
+			args: args{
+				svcConf: &serviceConfig{},
+				lbaasV2: &LbaasV2{
+					LoadBalancer{
+						opts: LoadBalancerOpts{
+							LBClasses: map[string]*LBClass{
+								"test-class": {
+									SubnetID:           "test-id",
```

Comment:
What do you think about putting something like `test-class-subnet-id` here, to make sure that the subnet ID is actually taken from the annotation?

Reply:
sound great! i've added it in